### DISCUSSION TITLE
How to expose different model fields in the frontend API?

### DISCUSSION THREAD

**Discussion by ieatmangos** [swellstores/community#24] [Help] [answered]
The frontend api is a subset of options available on the backend. How can I choose which fields to expose to the frontend api such that when grabbing the products, I also get their categories, category_id, category_index, or other fields only found in the backend api?

**Comment by rodrigoddalmeida** [comment, marked as answer]
Hi @ieatmangos, you can check that out in this guide: https://developers.swell.is/guides/opening-content-permissions

**Comment by vettloffah** [comment]
@rodrigoddalmeida that link is returning a 500 server error

**Reply by rodrigoddalmeida** [reply]
We are checking that out now.

**Reply by rodrigoddalmeida** [reply]
It should be back up. Thanks for the heads-up!

**Comment by ieatmangos** [comment]
Thanks for your response!

Im still trying to understand how this all works... "Making these permissions public will grant any user editing access to your store’s content models and fields."

What if I wanted to read only, not write. Is there a way to accomplish this without opening up edit access to any user?

Should I be using my pages/api folder (next.js) to run an instance of the backend, creating specific endpoints to call on the front end.. ie. a route called api/getAllProducts that is inclusive of their category (or whatever fields I want). Or if I should be making edits like you've said above and use swell.products.list({...}) in my web application?

**Comment by swellmike** [comment]
If you want `product.foo` to be readable via the public API, just add it to `products.fields`.  The `foo` field will not be writeable unless you also add it to `products.input.fields`.

```
"permissions": {
  "products": {
    "fields": ["foo"]
  }
}
```

**Reply by ieatmangos** [reply]
I've done this. And now when I GET my client keys, I see the new permissions added to the stack response, when using the console. However my front end application is still in the dark, nothing has changed. Do I need to update something in my frontend code base to sync to the new client key?

**Reply by ieatmangos** [reply]
I see from another discussion that I should simply be using the backend api and 'swell-node' within a server component or getStaticProps to get the full models.

**Reply by vettloffah** [reply]
yeah I don't think the answer solves the problem the OP is asking. You need to use the backend API to get more information on categories, in my experience. `swell-node` instead of `swell-js`, which means you need to get the data from the server instead of the front end in your application.

**Reply by ericingram** [reply]
We need to improve/simplify the frontend API permission model. But I think @vettloffah is right i this case
